### DISCUSSION TITLE
Next.js custom routes and now.json routes setup

### DISCUSSION THREAD

**Discussion by valpinkman** [vercel/vercel#4449] [Help]
Hello there,

Since the announcement of the v1 of now closing, I need to migrate my website to use the all new and shiny features :)

I'm having some issues tho, as my website has multiple languages, I used to have a custom server to handle thee different urls.

Right now though, I've converted the pages using the new `[pid].js` syntax for my dynamic routes, and I use the `now.json` **routes** to map all the different foreign languages urls to the next.js file system. 

I cannot however seem to setup now.json correctly for the dynamic routes.

Here is an example:

My directory would look like:
```
- pages
  - index.tsx
  - products
    - index.tsx
    - [pid].tsx
```

and my now.json

```
{
  ...
  "builds": [
      {
        "src": "package.json",
        "use": "@now/next"
      }
    ],
  "routes": [
    { "src": "/nos-produits/(?<product>[^/]+), "dest": "/products/$product" },
    { "src": "/nos-produits", "dest": "/products" },
  ]
}
```

With that setup, when I hit `/nos-produits/smoothies` for example, I get a 404. 
I'm not sure what I'm doing wrong, so if anyone could lend a hand that would be greatly appreciated.

Thanks

ps: I'm using `now dev` for my local development, with these scripts in package.json
```
{
   "scripts": {
      "dev": "next",
      "now-dev": "next",
      "now-build": "next build",
      "build": "next build",
      "start": "next start"
  }
}
```

and even with zero **routes** config in my now.json, it seems the route `/products/anything` does not event work. 

ps2: it seems to be a bug with `now dev` as `npm run dev` will actually match the routes 🙀

**Comment by mcsdevv** [comment]
Hello @valpinkman, from what you've said you shouldn't need to use any configuration at all.

You can pass the dynamic routing query to `[pid].tsx` and pick it up using the Next.js router.

**Reply by valpinkman** [reply]
Hmm, I would you map several languages to the same route then ?
 - `/our-products` => `pages/products.js`
- `/nos-produits` => `pages/products.js` 
 - `/our-products/smooties` => `pages/products/[pid].js`
- `/nos-produits/smooties` => `pages/products/[pid].js` 

I still haven't found a way... maybe I missed something in the nextjs router ?

Thanks anyway for the help :)

**Reply by valpinkman** [reply]
So, I can use `<Link href="..." as="...">`, however, it will not work on a page refresh (`/our-products` does not exist if I don't setup a `now` router)
